NetKet/Machine/py_machine.hpp

```diff
@@ -147,7 +147,7 @@ void AddMachineModule(py::module &m) {
            )EOF")
       .def("to_array",
            [](AbstractMachine &self) -> AbstractMachine::VectorType {
-             HilbertIndex hind(self.GetHilbert());
+             const auto &hind = self.GetHilbert().GetIndex();
              AbstractMachine::VectorType vals(hind.NStates());
 
              double maxlog = std::numeric_limits<double>::lowest();
```

Comment:
also these operations here would be vastly more elegant with c++ range iterators...

Reply:
The code in `to_array` also uses the index `i`, so this would not work well as a range-based for loop.
Still, with the commits I just pushed, one can write
```cpp
for (const auto state : hilbert) {
    // ...
}
```
to iterate over all configurations.